### DISCUSSION TITLE
Allow usage of operators in the filename of --form parameters for file uploads

### DISCUSSION THREAD

**Discussion by ghost** [curl/curl#6669] [Ideas]
Curl supports usage of `multipart/form-data` requests via `--form` parameters. There, the operator `@` can be used as an instruction to force the content part to be a file. The other operator `<` can be used to get the contents from a file and use it as an input for a text field content.

The following more complex usage of operators will work as expected:

```
curl --form "submit=<content.txt;headers=@headers.txt" https://www.example.com
curl --form "submit=<content.txt;headers=<headers.txt" https://www.example.com
curl --form "file=@content.png;headers=@headers.txt" https://www.example.com
curl --form "file=@content.png;headers=<headers.txt" https://www.example.com
```

However, these operators can not be used for the filename. Instead, the filename would always literally be `@filename.txt` or `<filename.txt`:

```
curl --form "file=@content.png;filename=@filename.txt" https://www.example.com
curl --form "file=@content.png;filename=<filename.txt" https://www.example.com
```

Worth noting is that using the operators with `type=` or `encoder=` an error is displayed. In the following, this is exemplarily shown with the `<` operator:

```
$ curl --form "submit=@content.png;type=<type.txt" https://www.example.com
Warning: Illegally formatted content-type field!
curl: option --form: is badly used here
curl: try 'curl --help' or 'curl --manual' for more information

$ curl --form "submit=@content.png;encoder=<encoding.txt" https://www.example.com
Warning: curl_mime_encoder failed!
curl: option --form: is badly used here
curl: try 'curl --help' or 'curl --manual' for more information
```

While allowing files for `type=` and `encoding=` is probably only interesting for modularisation, the `filename=` is quite interesting in terms of supplying a very large filename or file contents as a filename.

Maybe, it is even easier to maintain these parts if they all behave the same.

What do you think about this?

**Comment by bagder** [comment]
I wouldn't mind supporting `@` and `<` for all of those! Seems like a totally reasonable idea to me!
